packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -16,6 +16,32 @@ import {downloadOrLoadFile} from "../../util";
 import {IBeaconArgs} from "./options";
 import {defaultNetwork, IGlobalArgs} from "../../options/globalOptions";
 import {getGenesisFileUrl} from "../../networks";
+import {
+  mainnetWeakSubjectivityState,
+  praterWeakSubjectivityState,
+  pyrmontWeakSubjectivityState,
+} from "../weakSubjectivityState";
+
+async function initAndVerifyWeakSujectivityState(
+  args: IBeaconArgs & IGlobalArgs,
+  wsStateData: {
+    stateRoot: string;
+    ipfsPath: string;
+  },
+  initFunc: (pathOrUrl: string) => Promise<TreeBacked<allForks.BeaconState>>,
+  logger: ILogger
+): Promise<TreeBacked<allForks.BeaconState>> {
+  logger.info("Fetching weak subjectivity state from IPFS at " + args.ipfsGatewayUrl + wsStateData.ipfsPath);
+  const state = await initFunc(args.ipfsGatewayUrl + wsStateData.ipfsPath);
+  if (!state) {
+    throw new Error("Weak subjectivity state not found for network " + args.network);
+  }
+  // verify downloaded state against locally stored state root
+  if (toHexString(state.hashTreeRoot()) !== wsStateData.stateRoot) {
+    throw new Error("Unable to verify state root downloaded from IPFS");
```

Comment:
You could verify it but it is invalid. You should print the expect root and the actual root